ignite/distributed/auto.py

```diff
@@ -101,7 +104,7 @@ def auto_dataloader(dataset, **kwargs):
         kwargs["pin_memory"] = kwargs.get("pin_memory", "cuda" in idist.device().type)
 
     logger.info("Use data loader kwargs for dataset '{}': \n\t{}".format(repr(dataset)[:20].strip(), kwargs))
-    dataloader = DataLoader(dataset, **kwargs)
+    dataloader = DataLoader(dataset, **kwargs)  # type: Union[DataLoader, "_MpDeviceLoader"]
```

Comment:
Here it is DataLoader type without _MpDeviceLoader

Reply:
at this specific line you are right, but a few lines later it can be reassigned with `_MpDeviceLoader` and results in following error
```
ignite/distributed/auto.py:122: error: Incompatible types in assignment (expression has type "_MpDeviceLoader", variable has type "DataLoader[Any]")  [assignment]
```

Reply:
OK, what if we do something like 
```python
dataloader = DataLoader(dataset, **kwargs)  # type: DataLoader

if idist.has_xla_support and idist.backend() == idist_xla.XLA_TPU and world_size > 1:
    sampler = dataloader.sampler  # type: ignore[union-attr]
    mp_dataloader = mp_device_loader_cls(dataloader, idist.device())  # type: "_MpDeviceLoader"
    mp_dataloader.sampler = sampler  # type: ignore[attr-defined]
    return mp_dataloader

return dataloader
```

Reply:
yea, should work. let me try it.